### PR TITLE
Create the topology client only once during creation of the NFD master

### PR DESCRIPTION
**Pull request by cynepco3hahue**
Signed-off-by: Artyom Lukianov <alukiano@redhat.com>

### CODE REVIEW

File: pkg/nfd-master/nfd-master.go [hunk 1]
```diff
@@ -141,13 +142,13 @@ func NewNfdMaster(args Args) (NfdMaster, error) {
 	// Initialize Kubernetes API helpers
 	nfd.apihelper = apihelper.K8sHelpers{Kubeconfig: args.Kubeconfig}
 
-	clientConfig, err := rest.InClusterConfig()
+	restConfig, err := rest.InClusterConfig()
```
Comment (swatisehgal) [inline pkg/nfd-master/nfd-master.go:146]: Import for rest ("k8s.io/client-go/rest") has been removed above

File: pkg/nfd-master/nfd-master.go [hunk 2]
```diff
@@ -572,15 +573,6 @@ func addAnnotations(n *api.Node, annotations map[string]string) {
 	}
 }
 
-func updateMap(input map[string]int32) map[string]int {
```
Comment (swatisehgal) [inline pkg/nfd-master/nfd-master.go]: We shouldn't be removing this function as it is used [here](https://github.com/swatisehgal/node-feature-discovery/blob/rte/pkg/nfd-master/nfd-master.go#L599). Please refer to [this](https://github.com/swatisehgal/node-feature-discovery/pull/1#pullrequestreview-534225565) comment on why we needed this in the first place.

File: pkg/nfd-master/nfd-master.go [hunk 3]
```diff
@@ -604,21 +596,11 @@ func modifyCRD(topoUpdaterZones map[string]*topologypb.Zone) map[string]v1alpha1
 
 }
 
-func updateCRD(hostname string, tmpolicy []string, topoUpdaterZones map[string]*topologypb.Zone, namespace string) error {
+func (s *nodeTopologyServer) updateCRD(hostname string, tmpolicy []string, topoUpdaterZones map[string]*topologypb.Zone, namespace string) error {
 	log.Printf("Exporter Update called NodeResources is: %+v", topoUpdaterZones)
-
```
Comment (swatisehgal) [inline pkg/nfd-master/nfd-master.go:609]: Good catch! We are indeed getting the client twice. Thanks for the fix!

**Review by swatisehgal — CHANGES_REQUESTED** [review]
Just a few minor changes are needed but other than that LGTM!

File: pkg/nfd-master/nfd-master.go [hunk 4]
```diff
@@ -30,18 +30,19 @@ import (
...
+	"github.com/swatisehgal/topologyapi/pkg/apis/topology/v1alpha1"
+	topologyclientset "github.com/swatisehgal/topologyapi/pkg/generated/clientset/versioned"
 	"golang.org/x/net/context"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials"
 	"google.golang.org/grpc/peer"
 	api "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/client-go/rest"
```
Comment (swatisehgal) [inline pkg/nfd-master/nfd-master.go:44]: "k8s.io/client-go/rest" removed but used later

Comment (cynepco3hahue) [reply pkg/nfd-master/nfd-master.go:44]: yep it moved above, I updated imports again

File: pkg/nfd-master/nfd-master.go (hunk 1 above)
Comment (cynepco3hahue) [reply pkg/nfd-master/nfd-master.go:146]: it was moved above, I updated all imports

File: pkg/nfd-master/nfd-master.go (hunk 2 above)
Comment (cynepco3hahue) [reply pkg/nfd-master/nfd-master.go]: reverted

**Comment by cynepco3hahue** [comment]
@swatisehgal Updated

**Comment by swatisehgal** [comment]
LGTM! Thanks for the PR

**Pull request merged**